app/controllers/amr_heatmap_controller.rb

```diff
@@ -0,0 +1,39 @@
+class AmrHeatmapController < ApplicationController
+  include ReportHelper
+
+  before_action :admin_required
+
+  power :viewable_samples
+
+  # GET /amr_heatmap.json
+  # Return JSON information required to create a visualization,
+  # from submitted parameters
+  def amr_counts
+    samples = []
+    amr_data = []
+
+    params[:id].each do |param_id|
+      sample = current_power.viewable_samples.find(param_id)
+      assert_access
+      samples << sample
+    end
+
+    samples.each do |sample|
+      amr_counts = nil
+      pipeline_run = sample.first_pipeline_run
+      if pipeline_run
+        amr_state = pipeline_run.output_states.find_by(output: "amr_counts")
+        if amr_state.present? && amr_state.state == PipelineRun::STATUS_LOADED
+          amr_counts = pipeline_run.amr_counts
+        end
+      end
+      amr_data << {
+        sample_name: sample.name,
+        sample_id: sample.id,
+        amr_counts: amr_counts
+      }
+    end
+
+    render json: amr_data
```

Comment:
I debated the need to add a `missingSampleIds` field to the response, which explicitly lists the sample ids that we couldn't find. I don't think this is necessary right now (since we can infer on the front-end which sample ids weren't returned). We might want to do something like this if there are multiple reasons a sample can fail to fetch, and we want to surface the failure reason to the user.

Reply:
Sample ids that are invalid (for whatever reason) now just return the same object as valid ids, but with `sample_name` set to an empty string and `error` set to `"sample not found"`. I could undo it, but it works for now and I prefer it over failing silently.

Reply:
Sounds good.